source/common/network/socket_interface.h

```diff
@@ -52,5 +52,23 @@ static inline const SocketInterface* socketInterface(std::string name) {
 using SocketInterfaceSingleton = InjectableSingleton<SocketInterface>;
 using SocketInterfaceLoader = ScopedInjectableLoader<SocketInterface>;
 
+/**
+ * Create IoHandle for given address
+ * @param type type of socket to be requested
+ * @param addr address that is gleaned for address type, version and socket interface name
+ * @return @ref Network::IoHandlePtr that wraps the underlying socket file descriptor
+ */
+static inline IoHandlePtr ioHandleForAddr(Socket::Type type,
+                                          const Address::InstanceConstSharedPtr addr) {
+  auto sock_interface_name = addr->socketInterface();
+  if (!sock_interface_name.empty()) {
+    auto sock_interface = const_cast<SocketInterface*>(socketInterface(sock_interface_name));
```

Comment:
nit: this const_cast seems a bit off. Any reason to not just fix the socket method itself to be const?

Reply:
I thought about that too but `socket()` could end up changing the `SocketInterface` object. The alternative would be to remove const from `SocketInterface` and trust that callers won't do anything irresponsible.

Reply:
Why would it change the interface object? Does that happen in practice right now? Can we cross that bridge later if we come to it? If it really is a mutable thing I would just remove const.

Reply:
I was thinking that interface objects could be tracking the sockets, e.g., they could be building some form of socket descriptor map. We're not doing that now, as we're offloading everything to the kernel, but I did hit this particular issue with a custom socket interface implementation. But okay, let's cross that bridge later :-)